java/client/src/main/java/io/vitess/client/VTGateConnection.java

```diff
@@ -118,14 +119,11 @@ public VTGateConnection(RpcClient client, Query.ExecuteOptions options) {
      */
     public VTGateConnection(RpcClient client, String targetString, Query.ExecuteOptions options) {
         this.client = checkNotNull(client);
-        this.session = Session.newBuilder()
-                .setTargetString(null == targetString ? "" : targetString)
-                .setOptions(null == options ? Query.ExecuteOptions.newBuilder().build() : options)
-                .build();
+        this.globalSession = new VTSession(targetString, options);
```

Comment:
One concern about having a global session and allowing one to be passed in as well is that each response is modifying the passed in session, so the global one will quickly become outdated. If someone alternates between the two function types they may not realize.  Maybe we should have a separate `VTGateConnectionWithGlobalSession`  which delegates to VTGateConnection and only exposes non-with-session functions.